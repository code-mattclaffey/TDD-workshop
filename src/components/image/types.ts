import { GenericComponent } from '../../module';

export interface ImageInterface extends GenericComponent {
    src: string;
    alt: string;
}
