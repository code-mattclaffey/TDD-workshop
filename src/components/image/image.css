/* ------------------------------------
    #IMAGES
------------------------------------*/

img,
.e-img {
    height: auto;
    max-width: 100%;
}
