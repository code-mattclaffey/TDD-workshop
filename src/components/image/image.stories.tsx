import React from 'react';
import { Image } from '.';

export default {
    title: 'Image',
    componentSubtitle: '',
};

export const image = () => (
    <Image
        src="https://via.placeholder.com/400x300"
        alt="A placeholder image"
    />
);
