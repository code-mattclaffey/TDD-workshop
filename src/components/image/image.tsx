import React from 'react';
import classNames from 'classnames';
import { ImageInterface } from './types';

export const Image = ({
    additionalClassNames = '',
    src,
    alt,
    children,
    ...rest
}: ImageInterface) => {
    const classes = classNames(additionalClassNames);

    return <img className={classes} {...rest} src={src} alt={alt} />;
};
