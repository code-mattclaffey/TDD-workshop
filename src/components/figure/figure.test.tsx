import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Figure, FigCaption } from '.';
import { Image } from '../image';

afterEach(() => {
    cleanup();
});

describe('<Figure />', () => {
    it('should match Figure only snapshot', () => {
        const { container } = render(
            <Figure
                captionText="Hello World"
                additionalClassNames="c-figure"
            >
                <Image
                    src="https://via.placeholder.com/400x300"
                    alt="A placeholder image"
                />
            </Figure>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match Figure & FigCaption snapshot', () => {
        const { container } = render(
            <Figure additionalClassNames="c-figure">
                <Image
                    src="https://via.placeholder.com/400x300"
                    alt="A placeholder image"
                />
                <FigCaption additionalClassNames="c-figure__caption">
                    Hello World
                </FigCaption>
            </Figure>,
        );

        expect(container).toMatchSnapshot();
    });
});
