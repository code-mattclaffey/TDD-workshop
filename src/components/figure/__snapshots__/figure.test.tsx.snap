// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Figure /> should match Figure & FigCaption snapshot 1`] = `
<div>
  <figure
    class="e-figure c-figure"
  >
    <img
      alt="A placeholder image"
      class="e-img"
      src="https://via.placeholder.com/400x300"
    />
    <figcaption
      class="e-figure__caption c-figure__caption"
    >
      Hello World
    </figcaption>
  </figure>
</div>
`;

exports[`<Figure /> should match Figure only snapshot 1`] = `
<div>
  <figure
    class="e-figure c-figure"
  >
    <img
      alt="A placeholder image"
      class="e-img"
      src="https://via.placeholder.com/400x300"
    />
    <figcaption
      class="e-figure__caption"
    >
      Hello World
    </figcaption>
  </figure>
</div>
`;
